tests/components/api/test_init.py

```diff
@@ -569,11 +570,41 @@ async def test_event_stream_requires_admin(
     assert resp.status == HTTPStatus.UNAUTHORIZED
 
 
-async def test_states_view_filters(
+async def test_states(
     hass: HomeAssistant, mock_api_client: TestClient, hass_admin_user: MockUser
+) -> None:
+    """Test fetching all states as admin."""
+    hass.states.async_set("test.entity", "hello")
+    resp = await mock_api_client.get(const.URL_API_STATES)
+    assert resp.status == HTTPStatus.OK
+    json = await resp.json()
+    assert len(json) == 1
+    assert json[0]["entity_id"] == "test.entity"
+
+
+async def test_states_view_filters(
```

Comment:
This appears to test the case where the filter passes where the user has permissions, but i don't think I see an existing test where a test verifies that it filters out entities that the user does not have permissions for. This is an existing gap, but perhaps worth adding that while you're here? Approving either way.

Reply:
Its a bit confusing since the existing test was using a hass admin user, but this test is validating that the user cannot see the entity created with `hass.states.async_set("test.not_visible_entity", "invisible")` via 
```
    assert len(json) == 1
    assert json[0]["entity_id"] == "test.entity"`

```